app/views/unattended/provisioning_templates/snippet/preseed_kernel_options_autoinstall.erb

```diff
@@ -30,7 +35,14 @@ description: options for the kernel / preseed startup initialization
     options << "ip=[#{iface.ip6}]::[#{subnet6.gateway}]:[#{subnet6.mask}]:#{hostname}:#{iface.identifier}:none:[#{subnet6.dns_servers.join(']:[')}]"
   end
 
+  options << 'ramdisk_size=1500000'
+  options << 'fsck.mode=skip'
+  options << 'autoinstall'
+  options << "url=http://#{@preseed_server}#{image_path}"
+  options << 'cloud-config-url=/dev/null'
+  options << "ds=nocloud-net;s=http://#{foreman_request_addr}/userdata/#{mac ? mac + '/' : ''}"
```

Comment:
Adding the mac here results in more testing, since this introduces changes for the PXELinux and PXEGrub2 templates as well (even though we only want to add the iPXE template in this PR).

Therefore, I would recommend keeping this like before and creating a separate ticket for it.
Do you experience issues if the mac is not given in iPXE?

Reply:
It's only changing the URL so should make no difference to PXELinux or PXEGrub2 handling the option, as long as the correct template is rendered. 

The only query parameters `UserdataController` uses to lookup the host is the remote IP, and the mac in the URL. Therefore, for hosts assigned an IP through DHCP mac is the only way to find the right host. So my reasoning is that all templates need pass the mac to be able to handle DHCP hosts. 

Looking at `HostFinder.find_host_by_ip_or_mac`, if given both a remote IP and a mac it will only search by mac. At a guess that's what's wanted? If the provisioning NIC has been setup with a mac then that's probably put in there for this reason? OTOH, remote IP is somewhat more secure as it's harder to spoof.